File: app/javascript/editors/extensions/HorizontalRule.ts
// from https://github.com/afomera/richer-text.js/blob/main/src/editor/extensions/HorizontalRule.js

import HorizontalRule from '@tiptap/extension-horizontal-rule'

export default HorizontalRule.extend({
    renderHTML({HTMLAttributes}) {
        return ["div", {class: "richer-text-editor--hr"}, ["hr", HTMLAttributes]];
    },
});

File: app/business/schemas/tip_tap/nodes/horizontal_rule.rb
# frozen_string_literal: true

# https://github.com/ueberdosis/tiptap/blob/develop/packages/extension-horizontal-rule/src/horizontal-rule.ts

module Schemas
  module TipTap
    module Nodes
      class HorizontalRule < Schemas::TipTap::Node
        self.type_name = 'horizontalRule'

        def to_html
          tag.hr
        end

        def to_markdown(_context = Markdown::Context.root)
          '---'
        end
      end
    end
  end
end
